api/src/main/java/org/apache/iceberg/ManageSnapshots.java

```diff
@@ -48,6 +58,7 @@
    * @return this for method chaining
    * @throws IllegalArgumentException If the table has no old snapshot before the given timestamp
    */
-  Rollback toSnapshotAtTime(long timestampMillis);
+  ManageSnapshots rollbackAtTime(long timestampMillis);
```

Comment:
Can we rename this to `rollbackToTime`? I think that wording is a little more natural.

While we're changing this API, I think we should also be a bit more careful about behavior. Should this roll back to the snapshot that was current at the given time, or to the snapshot that was committed just before the given time that is an ancestor of the current state? For example, say I have snapshots A, B, and C committed in that order, while my table history is A, B, C, B. If I roll back to a time when C was current, should this use B (because C is not an ancestor of the current state) or should this use C because it was current at the time? I'm leaning toward using only snapshots that are ancestors of the current table state, but I'd like to hear other opinions on it.

Reply:
For history, the snapshot log wasn't there when I created the rollback operation, which is why we didn't choose a different behavior at the time.

Reply:
Current behavior is:  As per your example A, B, C, B .. if i rollback to a time that C was current (as in rollback to `C.timestampMillis()`  ) then B would be the latest snapshot that is older than that timestamp. 
I like this approach coz it protects against rounding off errors and is consistent.